tests/e2e/cucumber/step_definitions/app.files.ts

```diff
@@ -99,6 +99,52 @@ When(
   }
 )
 
+When(
+  /^"([^"]*)" creates public link to the following resource(s)? via the (sidebar panel|quick action)$/,
+  async function (
+    this: World,
+    stepUser: string,
+    _: string,
+    actionType: string,
+    stepTable: DataTable
+  ) {
+    const actor = this.actorsEnvironment.getActor({ id: stepUser })
+    const { publicLink: publicLinkPage } = new FilesPage({ actor })
+
+    const shareInfo = stepTable.hashes().reduce((acc, stepRow) => {
+      const { resource, name, role, dateOfExpiration, password } = stepRow
+      if (!acc[resource]) {
+        acc[resource] = { name: '', role: '', dateOfExpiration: '', password: '' }
+      }
+      acc[resource].name = name
+      acc[resource].role = role
+      acc[resource].dateOfExpiration = dateOfExpiration
+      acc[resource].password = password
+      return acc
+    }, {})
+
+    for (const folder of Object.keys(shareInfo)) {
+      await publicLinkPage.publicLinkResource({
+        folder,
+        name: shareInfo[folder].name,
+        role: shareInfo[folder].role,
+        dateOfExpiration: shareInfo[folder].dateOfExpiration,
+        password: shareInfo[folder].password,
+        via: actionType === 'quick action' ? 'QUICK_ACTION' : 'SIDEBAR_PANEL'
+      })
+    }
+  }
+)
+
+Then(
+  '{string} should see the public link',
```

Comment:
In future there will likely be scenarios where the user creates more than one public link on the same resource. We will want to assert that the expected number of public links are visible. Maybe we make this step be:
```suggestion
  '{string} should see 1 public link',
```

and allow `1`to be any number. And make the code find the element that contains all the public links for the resource, and count the number of public links found inside that element.